### PR TITLE
Edpm bm refactor

### PR DESCRIPTION
**Pull request by steveb**
Add growvols args to gen-edpm-baremetal-kustomize.sh
Align gen-edpm-baremetal-kustomize.sh files
    
(see commit messages for further context)

### CODE REVIEW

File: devsetup/scripts/gen-edpm-baremetal-kustomize.sh [hunk 1]
```diff
@@ -59,39 +59,42 @@ patches:
...
-      path: /spec/baremetalSetTemplate/ctlplaneInterface
-      value: enp1s0
-    - op: add
-      path: /spec/baremetalSetTemplate/provisioningInterface
-      value: ${PROVISIONING_INTERFACE}
+      value: ${EDPM_BMH_NAMESPACE}
     - op: add
-      path: /spec/env/0
-      value: {"name": "ANSIBLE_CALLBACKS_ENABLED", "value": "profile_tasks"}
+      path: /spec/nodeTemplate/networks
```
Comment (rabi) [inline devsetup/scripts/gen-edpm-baremetal-kustomize.sh:65]: We don't need this and can drop from both the scripts.

File: devsetup/scripts/gen-edpm-baremetal-kustomize.sh [hunk 2]
```diff
@@ -123,5 +120,32 @@ cat <<EOF >>kustomization.yaml
         namespace: ${NAMESPACE}
 EOF
 fi
+if [ "$EDPM_PROVISIONING_INTERFACE" != "" ]; then
```
Comment (rabi) [inline devsetup/scripts/gen-edpm-baremetal-kustomize.sh:123]: This is not optional when BMO deployed with CRC, unlike CBO and it's not in the sample spec[1]

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanenodeset_baremetal_with_ipam.yaml#L21

File: scripts/gen-edpm-baremetal-kustomize.sh [hunk 3]
```diff
@@ -50,7 +44,7 @@ resources:
 namespace: ${NAMESPACE}
 patches:
 - target:
-    kind: ${KIND}
+    kind: OpenStackDataPlaneNodeSet
```
Comment (rabi) [inline scripts/gen-edpm-baremetal-kustomize.sh:47]: Though we can hardcode this, it's set in https://github.com/openstack-k8s-operators/install_yamls/blob/main/Makefile#L573

File: devsetup/Makefile [hunk 4]
```diff
@@ -98,21 +101,25 @@ crc_attach_default_interface_cleanup: ## Detach default libvirt network from CRC
...
 edpm_baremetal: export EDPM_ANSIBLE_USER=${DATAPLANE_ANSIBLE_USER}
 edpm_baremetal: export BMAAS_NETWORK_NAME=${BM_NETWORK_NAME}
 edpm_baremetal: export BMAAS_NETWORK_IPADDRESS=${BM_NETWORK_IPADDRESS}
-edpm_baremetal: export EDPM_DEPLOY_STRATEGY_DEPLOY=${DATAPLANE_DEPLOY_STRATEGY_DEPLOY}
 edpm_baremetal: export BMAAS_INSTANCE_NAME_PREFIX=${BM_INSTANCE_NAME_PREFIX}
 edpm_baremetal: export EDPM_CHRONY_NTP_SERVER=${DATAPLANE_CHRONY_NTP_SERVER}
 edpm_baremetal: export EDPM_SSHD_ALLOWED_RANGES=${DATAPLANE_SSHD_ALLOWED_RANGES}
 edpm_baremetal: export EDPM_ROOT_PASSWORD_SECRET=${BM_ROOT_PASSWORD_SECRET}
 edpm_baremetal: export BMAAS_NODE_COUNT=${BM_NODE_COUNT}
 edpm_baremetal: export NODE_COUNT=${BM_NODE_COUNT}
```
Comment (rabi) [inline devsetup/Makefile:117]: We can remove the NODE_COUNT from here I think.

Comment (steveb) [reply devsetup/Makefile:117]: NODE_COUNT is used by vbm-setup.sh and edpm-compute-baremetal.sh.

File: devsetup/scripts/gen-edpm-baremetal-kustomize.sh (hunk 1 above)
Comment (steveb) [reply devsetup/scripts/gen-edpm-baremetal-kustomize.sh:65]: Done

File: scripts/gen-edpm-baremetal-kustomize.sh (hunk 3 above)
Comment (steveb) [reply scripts/gen-edpm-baremetal-kustomize.sh:47]: I have set this back to ${KIND}

File: devsetup/Makefile (hunk 4 above)
Comment (rabi) [reply devsetup/Makefile:118]: Yeah, we've now 3 vars for NODE_COUNT

**Review by rabi — APPROVED** [review]
/lgtm

**Review by hjensas — APPROVED** [review]
Looks good to me.

File: devsetup/Makefile (hunk 4 above)
Comment (steveb) [reply devsetup/Makefile:118]: I know its not great, I'd like to clean this up later

**Review by fao89 — APPROVED** [review]

**Pull request merged**